texar/torch/modules/connectors/connectors.py

```diff
@@ -128,19 +113,15 @@ def _mlp_transform(inputs: TensorStruct,
     """
     # Flatten inputs
     flat_input = nest.flatten(inputs)
-    if len(flat_input[0].size()) == 1:
-        batch_size = 1
-    else:
-        batch_size = flat_input[0].size(0)
-    flat_input = [x.view(-1, x.size(-1)) for x in flat_input]
+    flat_input = [x.view(x.size(0), -1) for x in flat_input]
     concat_input = torch.cat(flat_input, 1)
     # Get output dimension
     flat_output_size = nest.flatten(output_size)
 
     if isinstance(flat_output_size[0], torch.Size):
         size_list = [0] * len(flat_output_size)
         for (i, shape) in enumerate(flat_output_size):
-            size_list[i] = np.prod([dim for dim in shape])
+            size_list[i] = np.prod(shape)
     else:
         size_list = flat_output_size
```

Comment:
These few lines are exactly the same as a part of `_sum_output_size`. Why not take these out as a separate function (e.g. `_get_sizes`), and replace `_sum_output_size(xs)` with `sum(_get_sizes(xs))`?